pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -99,9 +98,21 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 			attrs.enableSriov = true
 			requireChange = true
 		}
-		if fwCurrent.linkType != EthLinkType && fwCurrent.linkType != PreconfiguredLinkType && fwCurrent.linkType != UknownLinkType {
-			attrs.linkType = EthLinkType
-			requireChange = true
+		if iface.LinkType != "" {
```

Comment:
So you defined a new field in the CRD, 
what is the expected behaviour if this field is not set ?

previously mellanox plugin configured the link type to Ethernet.
now it will simply do nothing. 
This change is not backward compatible right ? (which is something i think we want)

Reply:
Also, a separate method handling this logic will help with clarity.

Reply:
The behavior will be if there is no link type specified the plugin will assume it to be ETH

Reply:
> Also, a separate method handling this logic will help with clarity.

I still think this is desirable, but will not insist on it...